fili-core/src/test/groovy/com/yahoo/bard/webservice/web/handlers/SplitQueryRequestHandlerSpec.groovy

```diff
@@ -67,6 +69,28 @@ class SplitQueryRequestHandlerSpec extends Specification {
         1         | MONTH     | month
         12        | MONTH     | year
     }
+    
+    def "Handler skips splitting for all time grain"() {
+        groupByQuery.granularity >> timeGrain
+        groupByQuery.intervals >> [interval]
+        rc.numberOfIncoming >> new AtomicLong(1)
+        rc.numberOfOutgoing >> new AtomicLong(1)
+
+        when:
+        handler.handleRequest(rc, apiRequest, groupByQuery, response)
+
+        then:
+        (0) * groupByQuery.withAllIntervals(_) >> groupByQuerySplit
+        (1) * next.handleRequest(rc, apiRequest, groupByQuery, response)
+        
+        where:
+        timeGrain | interval
+        all       | week
+        all       | month
+        all       | month
+        all       | year
```

Comment:
Since the error we observed was that it was only incorrectly splitting for a disjoint interval, could we add a case that tests that condition as well?